ext/bcmath/libbcmath/src/div.c

```diff
@@ -31,219 +31,434 @@
+			sub = n2_vector[j] * quot_guess + borrow;
+			BC_VECTOR sub_low = sub % BC_VECTOR_BOUNDARY_NUM;
+			borrow = sub / BC_VECTOR_BOUNDARY_NUM;
+
+			if (n1_calc_bottom[j] >= sub_low) {
+				n1_calc_bottom[j] -= sub_low;
+			} else {
+				n1_calc_bottom[j] += BC_VECTOR_BOUNDARY_NUM - sub_low;
+				borrow++;
 			}
 		}
+		/* last digit sub */
+		sub = n2_vector[j] * quot_guess + borrow;
+		bool neg_remainder = n1_calc_bottom[j] < sub;
+		n1_calc_bottom[j] -= sub;
+
+		/* If the temporary quotient is too large, add back n2 */
+		BC_VECTOR carry = 0;
+		if (neg_remainder) {
+			quot_guess--;
+			for (j = 0; j < n2_arr_size - 1; j++) {
+				n1_calc_bottom[j] += n2_vector[j] + carry;
+				carry = n1_calc_bottom[j] / BC_VECTOR_BOUNDARY_NUM;
+				n1_calc_bottom[j] %= BC_VECTOR_BOUNDARY_NUM;
+			}
+			/* last add */
+			n1_calc_bottom[j] += n2_vector[j] + carry;
+		}
+
+		quot_vector[quot_top_index - i] = quot_guess;
+		div_carry = n1_vector[n1_top_index - i];
+		n1_vector[n1_top_index - i] = 0;
 	}
 }
 
+static void bc_do_div(char *n1, size_t n1_readable_len, size_t n1_bottom_extension, char *n2, size_t n2_len, bc_num *quot, size_t quot_len)
+{
+	size_t i;
+
+	size_t n2_arr_size = (n2_len + BC_VECTOR_SIZE - 1) / BC_VECTOR_SIZE;
+	size_t n1_arr_size = (n1_readable_len + n1_bottom_extension + BC_VECTOR_SIZE - 1) / BC_VECTOR_SIZE;
+	size_t quot_arr_size = n1_arr_size - n2_arr_size + 1;
+	size_t quot_real_arr_size = MIN(quot_arr_size, (quot_len + BC_VECTOR_SIZE - 1) / BC_VECTOR_SIZE);
+
+	BC_VECTOR *n1_vector = safe_emalloc(n1_arr_size + n2_arr_size + quot_arr_size, sizeof(BC_VECTOR), 0);
+	BC_VECTOR *n2_vector = n1_vector + n1_arr_size;
+	BC_VECTOR *quot_vector = n2_vector + n2_arr_size;
+
+	/* Fill with zeros and convert as many vector elements as needed */
+	size_t n1_vector_count = 0;
+	while (n1_bottom_extension >= BC_VECTOR_SIZE) {
+		n1_vector[n1_vector_count] = 0;
+		n1_bottom_extension -= BC_VECTOR_SIZE;
+		n1_vector_count++;
+	}
+
+	size_t n1_bottom_read_len = BC_VECTOR_SIZE - n1_bottom_extension;
+
+	size_t base;
+	size_t n1_read = 0;
+	if (n1_bottom_read_len < BC_VECTOR_SIZE) {
+		n1_read = MIN(n1_bottom_read_len, n1_readable_len);
+		base = POW_10_LUT[n1_bottom_extension];
+		n1_vector[n1_vector_count] = 0;
+		for (i = 0; i < n1_read; i++) {
+			n1_vector[n1_vector_count] += *n1 * base;
+			base *= BASE;
+			n1--;
+		}
+		n1_vector_count++;
+	}
+
+	/* Bulk convert n1 and n2 to vectors */
+	if (n1_readable_len > n1_read) {
+		bc_convert_to_vector(n1_vector + n1_vector_count, n1, n1_readable_len - n1_read);
+	}
+	bc_convert_to_vector(n2_vector, n2, n2_len);
+
+	/* Do the divide */
+	if (n2_arr_size == 1) {
+		bc_fast_div(n1_vector, n1_arr_size, n2_vector[0], quot_vector, quot_arr_size);
+	} else {
+		bc_standard_div(n1_vector, n1_arr_size, n2_vector, n2_arr_size, n2_len, quot_vector, quot_arr_size);
+	}
+
+	/* Convert to bc_num */
+	char *qptr = (*quot)->n_value;
+	char *qend = qptr + quot_len - 1;
+
+	i = 0;
```

Comment:
Useless line? The for loop initializes this to zero in the next line

Reply:
You are right, thx. I fixed it
https://github.com/php/php-src/pull/14660/commits/611fbb77b001366753e32a12872befe99b6d4164